integration-tests/src/test/java/com/datastax/oss/driver/core/cql/BoundStatementCcmIT.java

```diff
@@ -365,6 +366,44 @@ public void should_compute_routing_key_when_indices_randomly_distributed() {
     }
   }
 
+  @Test
+  public void should_set_all_occurrence_of_variable() {
+    CqlSession session = sessionRule.session();
+    PreparedStatement ps = session.prepare("INSERT INTO test3 (pk1, pk2, v) VALUES (:i, :i, :i)");
+
+    CqlIdentifier id = CqlIdentifier.fromCql("i");
+    ColumnDefinitions variableDefinitions = ps.getVariableDefinitions();
+    assertThat(variableDefinitions.allIndicesOf(id)).containsExactly(0, 1, 2);
+
+    should_set_all_occurrences_of_variable(ps.bind().setInt(id, 12));
+    should_set_all_occurrences_of_variable(ps.boundStatementBuilder().setInt(id, 12).build());
+  }
+
+  private void should_set_all_occurrences_of_variable(BoundStatement bs) {
+    assertThat(bs.getInt(0)).isEqualTo(12);
+    assertThat(bs.getInt(1)).isEqualTo(12);
+    assertThat(bs.getInt(2)).isEqualTo(12);
+
+    // Nothing should be shared internally (this would be a bug if the client later retrieves a
+    // buffer with getBytesUnsafe and modifies it)
```

Comment:
I want emphasize this. I think reusing the same buffer would be a mistake, but just to check everyone agrees.

Reply:
Agreed, while using names or ids you would always modify the values of all occurrences, when using indices you still can modify just one of the occurrences.